src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -355,6 +369,8 @@ public void StartProfileOptimization(string profile)
 #endif // FEATURE_MULTICOREJI
         }
         
+        public event ResolveEventHandler AssemblyResolve;
```

Comment:
The add/remove code generated by the C# compiler is thread safe, it looks something like this (decompiled from AppDomain.AssemblyLoad):

```C#
AssemblyLoadEventHandler handler2;
AssemblyLoadEventHandler assemblyLoad = this.AssemblyLoad;

do {
    handler2 = assemblyLoad;
    AssemblyLoadEventHandler handler3 = (AssemblyLoadEventHandler) Delegate.Combine(handler2, value);
    assemblyLoad = Interlocked.CompareExchange<AssemblyLoadEventHandler>(ref this.AssemblyLoad, handler3, handler2);
}
while (assemblyLoad != handler2);
```